src/components/Header.tsx

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-console */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import {
+  FC,
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
+import classNames from 'classnames';
+
+import { ActiveTodoData } from '../types/ActiveTodoData';
+import { CustomError } from '../types/CustomError';
+import { NewTodo, Todo } from '../types/Todo';
+import { postTodo } from '../api/todos';
+import { initData } from '../constants/initData';
+
+type Props = {
+  tempTodo: Todo | null,
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  activeTodoData: ActiveTodoData,
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>,
+  setError: (newError: CustomError, delay?: number) => void,
+};
+
+export const Header: FC<Props> = ({
+  tempTodo,
+  setTodos,
+  activeTodoData,
+  setTempTodo,
+  setError,
+}) => {
+  const [newTodo, setNewTodo] = useState<NewTodo>(initData.newTodo);
+  const [inputDisabled, setInputDisabled] = useState<boolean>(false);
+
+  const focusInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (focusInput.current) {
+      focusInput.current.focus();
+    }
+  }, [tempTodo]);
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (!newTodo.title.length) {
+      setError(CustomError.emptyTitle, 3000);
+    } else {
+      setInputDisabled(true);
+      setTempTodo({ ...newTodo, id: 0 });
+
```

Comment:
you can apply early return pattern here
```suggestion
      return setError(CustomError.emptyTitle, 3000);
    } 
    
    setInputDisabled(true);
    setTempTodo({ ...newTodo, id: 0 });

```